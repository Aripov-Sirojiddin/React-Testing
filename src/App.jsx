import { useState } from "react";
import "./App.css";

function App() {
  const [headingText, setHeadingText] = useState("Hello lads!");

  function handleOnClick() {
    setHeadingText("Good-bye lads!");
  }
  return (
    <>
      <h1>{headingText}</h1>
      <button onClick={handleOnClick}>Click to change heading</button>
    </>
  );
}

export default App;
