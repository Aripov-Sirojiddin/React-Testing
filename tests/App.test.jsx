import { render, screen } from "@testing-library/react";

import App from "../src/App.jsx";
import { expect } from "vitest";
import userEvent from "@testing-library/user-event";

describe("App Component", () => {
  it('renders headline as "Hello lads!"', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });

  it('changes the headline to "Good-bye lads!" on button click', async () => {
    const user = userEvent.setup();

    render(<App />);
    const button = screen.getByRole("button", {
      name: "Click to change heading",
    });

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch("Good-bye lads!");
  });
});
