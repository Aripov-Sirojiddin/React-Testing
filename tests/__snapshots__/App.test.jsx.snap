// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > renders headline as "Hello lads!" 1`] = `
<div>
  <h1>
    Hello lads!
  </h1>
  <button>
    Click to change heading
  </button>
</div>
`;
